### PR TITLE
[SAGE-352] Grid add fixed column width

### PR DESCRIPTION
**Pull request by QuintonJason**
## Description

- [x] add ability for the Grid to have a fixed width column. The parent grid should allow for the natural width of the children to determine the width

## Screenshots

variableWidth

## Testing in `sage-lib`

Go the Grid views and resize the browser to verify that `Variable Width` variation works as expected:
- [Rails](http://localhost:4000/pages/patterns/grid)
- [React](http://localhost:4100/?path=/story/sage-grid--fixed-width)

## Testing in `kajabi-products`

1. (**LOW**) Adds new `Grid` property. Has no affect in the app

## Related

Closes [SAGE-352](https://kajabi.atlassian.net/browse/SAGE-352)

### CODE REVIEW

File: packages/sage-react/lib/mocks/product-creation-wizard/components/Root.jsx [hunk 1]
```diff
@@ -114,7 +114,7 @@ export const Root = () => {
             <Grid.Col size={4} small={12} medium={5} large={4}>
               {renderStep()}
             </Grid.Col>
-            <Grid.Col size={8} small={0} medium={7} large={8}>
+            <Grid.Col variableWidth='225px'>
```
Comment (pixelflips) [inline packages/sage-react/lib/mocks/product-creation-wizard/components/Root.jsx]: Linter is yelling here.

**Review by pixelflips — CHANGES_REQUESTED** [review]
A couple of nitpicks to address but overall looking good so far.

File: packages/sage-react/lib/mocks/product-creation-wizard/components/Root.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (QuintonJason) [reply packages/sage-react/lib/mocks/product-creation-wizard/components/Root.jsx]: @philschanely Hey hey. I like `fixedWidth`. I did about 4 find replaces struggling to find something. You suggestion is solid. The part being passed in is wrong. I need to update. That value should be a `Boolean`

File: docs/lib/sage_rails/app/sage_components/sage_grid_col.rb [hunk 2]
```diff
@@ -2,5 +2,6 @@ class SageGridCol < SageComponent
   set_attribute_schema({
     breakpoint: [:optional, Symbol, String],
     size: [:optional, String, Integer],
+    variableWidth: [:optional, TrueClass]
```
[the remark this answers is not in the document]
Comment (QuintonJason) [reply docs/lib/sage_rails/app/sage_components/sage_grid_col.rb:5]: Great callout! Switching between rails and react 🙃

File: packages/sage-react/lib/mocks/product-creation-wizard/components/Root.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (QuintonJason) [reply packages/sage-react/lib/mocks/product-creation-wizard/components/Root.jsx]: @philschanely pushed

File: packages/sage-assets/lib/stylesheets/layout/_grid.scss [hunk 3]
```diff
@@ -86,18 +86,32 @@ $-grid-breakpoint-xl: sage-breakpoint(xl-min);
...
+  flex-grow: 1;
+  flex-basis: 0;
+  max-width: 100%;
+  padding: 0 calc(#{$-grid-gap} / 2);
+}
+
+.sage-col--auto {
+  flex: 0 0 auto;
+  width: auto;
+  max-width: none;
```
Comment (pixelflips) [inline packages/sage-assets/lib/stylesheets/layout/_grid.scss:105]: TIL max-width can be none!

**Review by pixelflips — APPROVED** [review]
LGTM and behaving as expected. Nice work! 👍🏼

**Comment by QuintonJason** [comment]
Closing in favor of this PR, #1375

**Pull request closed without merging**